src/reader/tf_reader.py

```diff
@@ -73,38 +66,44 @@ def _tf_parse_function(self, serialized):
                 'image': tf.io.FixedLenFeature([], tf.string),
                 'size': tf.io.FixedLenFeature([], tf.int64)
             }
-        return tf.io.parse_example(serialized=serialized, features=features)
+        parsed_example = tf.io.parse_example(serialized=serialized, features=features)
+        # Get the image as raw bytes.
+        image_raw = parsed_example['image']
+        dimension = tf.cast(parsed_example['size'], tf.int32).numpy()
+        # Decode the raw bytes so it becomes a tensor with type.
+        image_tensor = tf.io.decode_raw(image_raw, tf.float64)
+        size = dimension * dimension
+        dlp.update(image_size=size)
+        # image_tensor = tf.io.decode_image(image_raw)
+        resized_image = tf.reshape(image_tensor, [dimension, dimension])
```

Comment:
We have to have a similar logic here not to reshape but simulate preprocess and return resized image.

Reply:
I returned a tf.convert_to_tensor(resized_image)